handler.go

```diff
@@ -89,6 +89,55 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	h.mux.ServeHTTP(w, r)
 }
 
+// makeUserRequiredHandler takes a custom handler and returns a standard handler, ensuring that
+// the custom handler is only called with a authenticated user.
+func (h *Handler) makeUserRequiredHandler(fn func(http.ResponseWriter, *http.Request, *User)) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		q := r.URL.Query()
+		username, password := q.Get("u"), q.Get("p")
```

Comment:
Agreed -- we should do both. I'll flesh out the implementation more, and will include suppotr for Basic Auth.